frontend/web/components/pages/EnvironmentSettingsPage.tsx

```diff
@@ -0,0 +1,891 @@
+
+  // TODO: types
+  const getEnvironment = useCallback(async () => {
+    const env = ProjectStore?.getEnvs()?.find(
+      (env: Environment) => env.api_key === match.params.environmentId,
+    )
+    setCurrentEnv(env)
+
+    const roles = await getRoles(
+      store,
+      { organisation_id: AccountStore.getOrganisation().id },
+      { forceRefetch: true },
+    )
+    if (!roles?.data?.results?.length) return
+
+    const roleEnvironmentPermissions = await getRoleEnvironmentPermissions(
+      store,
+      {
+        env_id: env.id,
+        organisation_id: AccountStore.getOrganisation().id,
+        role_id: roles.data.results[0].id,
+      },
+      { forceRefetch: true },
+    )
+
+    const matchingItems = roles.data.results.filter((item1) =>
+      roleEnvironmentPermissions.data.results.some((item2) => item2.role === item1.id),
+    )
+    setRoles(matchingItems)
+
+
+    if (Utils.getPlansPermission('METADATA')) {
+      const supportedContentType = await getSupportedContentType(getStore(), {
+        organisation_id: AccountStore.getOrganisation().id,
+      })
+      const environmentContentType = Utils.getContentType(
+        supportedContentType.data,
+        'model',
+        'environment',
+      )
+      setEnvironmentContentType(environmentContentType)
+    }
+
+    await getWebhooks()
+  }, [match.params.environmentId, getWebhooks, store])
+
+  useEffect(() => {
+    AppActions.getProject(match.params.projectId)
+  }, [match.params.projectId])
+
+  useEffect(() => {
+    getEnvironment()
+  }, [match.params.environmentId, getEnvironment])
+
+  useEffect(() => {
+    API.trackPage(Constants.pages.ENVIRONMENT_SETTINGS)
+    getEnvironment()
+    getWebhooks()
+  }, [getEnvironment, getWebhooks])
+
+  const onSave = () => {
+    toast('Environment Saved')
+  }
+
+  const onRemove = () => {
+    toast('Your project has been removed')
+    router.history.replace(Utils.getOrganisationHomePage())
+  }
+
+  const confirmRemove = (environment: Environment, callback: () => void) => {
+    openModal(
+      'Remove Environment',
+      <ConfirmRemoveEnvironment environment={environment} cb={callback} />,
+      'p-0',
+    )
+  }
+
+  const onRemoveEnvironment = (environment: Environment) => {
+    const envs = ProjectStore.getEnvs() as Environment[] | null
+    if (envs && envs?.length > 0) {
+      router.history.replace(
+        `/project/${match.params.projectId}/environment` +
+        `/${envs[0].api_key}/features`,
+      )
+    } else {
+      router.history.replace(
+        `/project/${match.params.projectId}/environment/create`,
+      )
+    }
+    toast(
+      <div>
+        Removed Environment: <strong>{environment.name}</strong>
+      </div>,
+    )
+  }
+
+  const updateCurrentEnv = (newEnv: Partial<Environment> = {}, shouldSaveUpdate?: boolean, isDirtyDisabled?: boolean) => {
+    if (!isDirtyDisabled) {
```

Comment:
This is a flag to avoid "dirtying" the form. It was necessary for the feature versioning toggle (conflict with 2 different keys `enabledFeatureVersioning`, `use_v2_feature_versioning`).

if you have some more context on these 2 keys I can give it another look :)

Reply:
Then isDirty is set to true, on saving it's resetted